docs/003-markers.md

```diff
@@ -0,0 +1,441 @@
+* future markers (FMs)
+
+PMs and FMs are used to determine whether a message is in the past cone of the other, and FMs also help us efficiently estimate the approval weight of a message.
+
+#### StructureDetails Structure
+StructureDetails is a structure that will be in the message metadata containing marker information.
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>Rank</td>
+        <td>uint64</td>
+        <td>The rank of the message.</td>
+    </tr>
+    <tr>
+        <td>IsPastMarker</td>
+        <td>bool</td>
+        <td>A flag to indicate whether a message is a marker.</td>
+    </tr>
+    <tr>
+        <td>PastMarkers</td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Markers</summary>
+                <table>
+                    <tr>
+                        <th>Name</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>markers</td>
+                        <td>map[SequenceID]Index</td>
+                        <td>A list of PMs from different marker sequences.</td>
+                    </tr>
+                </table>
+            </details>
+        </td>
+    </tr>
+    <tr>
+        <td>FutureMarkers</td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Markers</summary>
+                <table>
+                    <tr>
+                        <th>Name</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>markers</td>
+                        <td>map[SequenceID]Index</td>
+                        <td>A list of FMs from different marker sequences.</td>
+                    </tr>
+                </table>
+            </details>
+        </td>
+    </tr>
+</table>
+
+##### Past Markers
+A PM of a marker is the marker itself only.
+A PM of general messages is inherited from its **strong** parents, with 2 steps:
+1. keep the nearest markers (i.e., the marker with the biggest *`MI`* ) of each marker sequence, 
+2. remove those that have been referenced by other markers in the same set. 
+
+##### Future Markers
+The FM is empty at start and gets updated when the new marker directly or indirectly references it. The FMs propagation ends if:
+1. the FM of a message includes a previous marker of the same marker sequence;
+2. the message is the first message we saw in the different marker sequence, we update the FM of that first message only.
+
+The figure below shows an example of applying markers to the Tangle with 1 marker sequence only:
+![](https://i.imgur.com/RluZWCJ.png)
+
+The yellow messages are markers with identifiers: *`[0,1]`* and *`[0,2]`*. Both markers are in *`MS`* *`0`* with *`MI`* *`1`* and *`2`* respectively.
+
```

Comment:
Add " For a more complete but complex example, including multiple sequences refer to the next figure."